lib/srv/db/proxy_test.go

```diff
@@ -54,6 +57,31 @@ func TestProxyProtocolPostgres(t *testing.T) {
 			psql, err := testCtx.postgresClientWithAddr(ctx, proxy.Address(), "alice", "postgres", "postgres", "postgres")
 			require.NoError(t, err)
 			require.NoError(t, psql.Close(ctx))
+
+			// Check that we can send a GSSEncRequest and get an 'N' response indicated server does not accept
+			// GSS encryption.
+			conn, err := net.Dial("tcp", proxy.Address())
+			require.NoError(t, err)
+			defer func() {
+				require.NoError(t, conn.Close())
+			}()
+			req := pgproto3.GSSEncRequest{}
+			msg := req.Encode(nil)
+			bytesWritten, err := conn.Write(msg)
+			require.NoError(t, err)
+			require.Equal(t, len(msg), bytesWritten)
+			buf := make([]byte, 512)
+			bytesRead, err := conn.Read(buf)
+			require.NoError(t, err)
+			require.Equal(t, []byte("N"), buf[:bytesRead], "server should respond with 'N' to indicate GSS encryption is not supported'")
+			sslReq := pgproto3.SSLRequest{}
+			msg = sslReq.Encode(nil)
+			bytesWritten, err = conn.Write(msg)
+			require.NoError(t, err, "server should not have closed conn after rejecting GSS")
+			require.Equal(t, len(msg), bytesWritten)
+			bytesRead, err = conn.Read(buf)
+			require.NoError(t, err)
+			require.Equal(t, []byte("S"), buf[:bytesRead], "server should respond with 'S' to indicate SSL encryption is supported'")
```

Comment:
nit: can we extract this GSS check to a separate function ?

Also from the test desc: 
> TestProxyProtocolPostgres ensures that clients can successfully connect to a
// Postgres database when Teleport is running behind a proxy that sends a proxy
// line.

Can we consider adding a new test instead of extending current test ?

Reply:
Will address this after fixing what we suggested in the other repo.

Reply:
Refactored into a (substantial) test that covers all code paths in `handleStartup`. I also made the tests run in parallel which sped things up substantially for tests running `lib/srv/db/proxy_test.go` from 15s runtime to 4.5 seconds locally. All tests still passed and I ran with the race detector on.